src/server/api/user.ts

```diff
@@ -80,6 +86,19 @@ function validateUrls(
     return
   }
 
+  // An upload request must contain a file
+  if (isFile) {
+    if (!req.files?.file) {
+      res.badRequest(jsonMessage('Missing file to upload.'))
+      return
+    }
+    const { isValidS3Shortlink } = container.get<S3Interface>(DependencyIds.s3)
```

Comment:
resolve dependencies during module initialisation at the top of the file, instead of during runtime.